### PR TITLE
Add method_missing to nginx_conf

### PR DESCRIPTION
**Pull request by rgeissert**
## Description
Add `method_missing` methods to greatly simplify the use of the resource. At present one needs to access the parameters directly from `params` and then flatten the result, which makes this resource particularly hard to use.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.

Aha! Link: https://chef.aha.io/features/SH-385

### CODE REVIEW

File: lib/inspec/resources/nginx_conf.rb
Comment (james-stocks) [inline lib/inspec/resources/nginx_conf.rb:57]: Please also define `respond_to_missing?` otherwise some weird issues can occur. 
https://github.com/inspec/inspec/issues/4591 was a previous issue that shows the errors that might occur.
https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding describes what to define and why.

**Comment by james-stocks** [comment]
Thanks very much for this @rgeissert - this is a great improvement. I left one comment about the implementation of `method_missing`

I also opened docs issue https://github.com/inspec/inspec/issues/5293 for us to consider later, because the existing example of defining ruby variables instead of using `its` is bad practice.

File: lib/inspec/resources/nginx_conf.rb
Comment (rgeissert) [reply lib/inspec/resources/nginx_conf.rb:57]: I just added them in d31e8e2f0. Note that almost none of the other resources implementing `method_missing` implement `respond_to_missing?`, and from what I noticed from tests they should be cautious and also exclude `to_` methods, and possibly others.

File: docs-chef-io/content/inspec/resources/nginx_conf.md
Comment (mjingle) [inline docs-chef-io/content/inspec/resources/nginx_conf.md]:
```suggestion
The `its` syntax allows for a more descriptive block and is available in the `nginx_conf`, `nginx_conf.http.entries`, and `nginx_conf.http.servers` resources.
```

File: docs-chef-io/content/inspec/resources/nginx_conf.md
Comment (mjingle) [inline docs-chef-io/content/inspec/resources/nginx_conf.md]:
```suggestion
    # Or when using `its` syntax
```

**Comment by mjingle** [comment]
Hi @rgeissert, Thank you for including documentation with your resource update! Because this is a PR from a remote fork, I have left two comments with suggested changes. Please implement these changes for this PR to have docs-team approval.

File: docs-chef-io/content/inspec/resources/nginx_conf.md
Comment (rgeissert) [reply docs-chef-io/content/inspec/resources/nginx_conf.md]: Applied in fbd5d3f.

File: docs-chef-io/content/inspec/resources/nginx_conf.md
Comment (rgeissert) [reply docs-chef-io/content/inspec/resources/nginx_conf.md]: Applied in fbd5d3f.

**Comment by rgeissert** [comment]
Hi @mjingle , I pushed the modifications you proposed. The verify build failed but AFAICS it isn't related to the changes.

**Comment by mjingle** [comment]
@rgeissert No worries. This PR will need engineering approval and the pipeline to be unblocked before merging, but you're good to go from a docs perspective. Thank you again for updating the documentation!

**Review by mjingle — APPROVED** [review]

**Comment by rgeissert** [comment]
@james-stocks I believe that the failed tests are unrelated to these changes. Could they be re-run?

**Comment by Schwad** [comment]
Hi @rgeissert - thanks much again for this and all your work! :) Once you review/address the pipelines breakages, could you @ me here and I'll come in so we can get this formally approved/merged? :)

**Review by clintoncwolfe — APPROVED** [review]

**Comment by rgeissert** [comment]
@Schwad after rebasing the only failures left appear to be due to a broken windows pipeline. It LGT otherwise.

**Pull request merged**
